modules/apps/journal/journal-web/src/main/java/com/liferay/journal/web/internal/info/item/renderer/JournalArticleDDMTemplateInfoItemTemplatedRenderer.java

```diff
@@ -94,20 +96,19 @@ public List<InfoItemRendererTemplate> getInfoItemRendererTemplates(
 					GetterUtil.getLong(classTypeKey)));
 		}
 
-		Stream<DDMStructure> stream = ddmStructures.stream();
+		for (List<DDMTemplate> ddmTemplates :
+				TransformUtil.transform(
+					ddmStructures, DDMStructure::getTemplates)) {
 
-		return stream.flatMap(
-			ddmStructure -> {
-				List<DDMTemplate> ddmTemplates = ddmStructure.getTemplates();
+			infoItemRendererTemplates.addAll(
+				TransformUtil.transform(
```

Comment:
@JanisZhang From performance point of view, it's better to just use 2-layer for-loop here. Please change and resend, thanks.